src/components/sign-in/sign-in.tsx

```diff
@@ -1,52 +1,29 @@
-import { Fragment, useState } from 'react';
-import { ethers } from 'ethers';
-import WalletConnectProvider from '@walletconnect/web3-provider';
-import Web3Modal from 'web3modal';
+import { Fragment } from 'react';
 import classNames from 'classnames';
-import { initialChainData, getNetworkData, useChainData, SettableChainData } from '../../chain-data';
+import { Address, useDisconnect, useEnsName } from 'wagmi';
+import { useChainData } from '../../chain-data';
 import { abbrStr } from '../../chain-data/helpers';
 import Button from '../../components/button';
+import ConnectButton from '../connect-button';
 import { Modal as GenericModal } from '../../components/modal';
 import Dropdown, { DropdownMenu, DropdownMenuItem } from '../../components/dropdown';
 import styles from './sign-in.module.scss';
 import globalStyles from '../../styles/global-styles.module.scss';
-import * as notifications from '../../components/notifications';
-import { images, messages } from '../../utils';
-import { SUPPORTED_NETWORKS, WALLET_CONNECT_RPC_PROVIDERS, useProviderSubscriptions } from '../../contracts';
-import { convertToEnsName } from '../../logic/proposals/encoding/ens-name';
-import { go } from '@api3/promise-utils';
+import { images } from '../../utils';
+import { SUPPORTED_NETWORKS, useProviderSubscriptions } from '../../contracts';
 
 type Props = {
   dark?: boolean;
   position: 'mobileMenu' | 'navigation';
 };
 
 const ConnectedStatus = ({ dark, position }: Props) => {
-  const { provider, setChainData, networkName, userAccount, availableAccounts, userAccountName } = useChainData();
+  const { networkName, userAccount } = useChainData();
 
-  const [changeAccountOpen, setChangeAccountOpen] = useState(false);
-  const openModal = () => setChangeAccountOpen(true);
-  const closeModal = () => setChangeAccountOpen(false);
+  const { disconnect } = useDisconnect();
+  const handleDisconnect = () => disconnect();
 
-  const onAccountChange = (account: string) => async () => {
-    if (!provider) return;
-
-    setChainData('Change user account', {
-      userAccount: account,
-      userAccountName: await convertToEnsName(provider, userAccount),
-      signer: provider.getSigner(account),
-    });
-  };
-
-  const onDisconnect = () => {
-    if (provider) {
-      const externalProvider: any = provider.provider;
-      if (typeof externalProvider.close === 'function') {
-        externalProvider.close();
-      }
-    }
-    setChainData('User disconnected', initialChainData);
-  };
+  const { data: userAccountName } = useEnsName({ address: userAccount as Address });
```

Comment:
Personal remark: I found it quite annoying how they strongly typed the `Address`.

Reply:
Yeah I also found that a bit overkill.